docs_src/design/adr/core/0026-Core-Data-Cache.md

```diff
@@ -0,0 +1,85 @@
+    - if there are more than N readings in core data, then delete the oldest readings, but keep the youngest N readings
+    - N must be > 0 (N = 1 means keep only the latest reading)
+    - this delete option will be triggered via API either:
+        - manually by the adopter
+        - or by the scheduling service (the adopter would set up a new IntervalAction with the scheduling service)
+
+- Delete readings to keep the database at a certain size (aka - the **Tony Delete**; clean readings to preserve the number of readings in the database)
+    - API:  none.  This operation is not visible via REST API
+    - this operation, when called upon internally by core data, will remove readings (from oldest to newest) to keep the core data database within a specified size
+    - a configuration setting (`CacheDelete` in configuration) will determine how readings will be removed by this operation:
+        - option A (`CacheDelete="OnEach"`): remove one reading (the oldest reading) each time a new reading gets added
+            - this operation would be called as part of the core data add event/reading operations
+            - this option causes core data to operate like a real cache - keeping only the latest reading
+        - option B (`CacheDelete="OnCount"`): remove a specified number (`DeleteSize` in configuration) of readings when the number of readings exceeds a specified limit (`CacheSize` in configuration)
+            - this operation would be called as part of the core data add event/reading operations
+            - this option causes core data to keep within a size range (between a max and min number of readings) 
+            - examples: 
+                - if there are 100 readings in the database and the `CacheSize` of the database readings is set to 100, then when the next reading is added to core data, a delete of the latest `DeleteSize` readings are removed (starting with the oldest readings) from core data.
+                - if there are 80 readings in the database and the `CacheSize` of the database readings is set to 100, then when the next reading is added to core data, no delete occurs.  The reading gets added normally and the number of readings in the database is now 81.
+            - this option reduces the add/delete thrashing.  It only deletes when the max size is hit or exceeded
+
+!!! Question
+        - On the Eaton delete - do we care to keep N readings per device or device resource?  Or we are just going to delete until we have only N readings and we don't care which ones we delete?
+
+        - On the Tony delete option A, given the way we specified this, there would only be one reading in core data at all times.  Do we mean to say that we only want to keep one reading of each device or device/device resource?  Or should we first allow some arbitrary number of readings to be built up and then start deleting readings one-for-one?
+
+        - Is the configuration for core data "cache" Writable or static?
```

Comment:
I think we would be OK with static; essentially it would be up to the product development team, looking at their storage resources and data generation, to gauge what would be good choices.
But the cost to make it Writable instead is pretty tiny.